gallery/tutorials/pipeline_demo.py

```diff
@@ -125,60 +122,78 @@
 # with noise and CTF corruption
 src.images[0:10].show()
 
+# %%
+# Image Preprocessing
+# -------------------
+# We apply some image preprocessing techniques to prepare the
+# the images for denoising via Class Averaging.
+
+# %%
+# Downsampling
+# ------------
+# We downsample the images to remove high frequency noise and improve the
+# efficiency of subsequent pipeline stages. Metadata such as pixel size is
+# scaled accordingly to correspond correctly with the image resolution.
+
+src = src.downsample(res)
+src.images[:10].show()
 
 # %%
 # CTF Correction
 # --------------
 # We apply ``phase_flip()`` to correct for CTF effects.
 
 src = src.phase_flip()
+src.images[:10].show()
+
+# %%
+# Normalize Background
+# --------------------
+# We apply ``normalize_background()`` to prepare the image class averaging.
+
+src = src.normalize_background()
+src.images[:10].show()
 
 # %%
-# Cache
-# -----
+# Caching
+# -------
 # We apply ``cache`` to store the results of the ``ImageSource``
 # pipeline up to this point.  This is optional, but can provide
 # benefit when used intently on machines with adequate memory.
-
+# Since the remaining preprocessing steps, whitening and contrast
```

Comment:
this is incorrect. You have it backwards.  The cache is in the wrong place to be effective.  Also you will need/want cache at multiple places here... I'm surprised it's not dramatically slower, but I guess it is a pretty small problem.

Reply:
Ok. Now caching the Simulation and each stage of preprocessing. Added a comment about caching the Simulation and removed the `Caching` section.